_pytest/python.py

```diff
@@ -889,13 +892,22 @@ def parametrize(self, argnames, argvalues, indirect=False, ids=None,
         if scope is None:
             scope = "function"
         scopenum = scopes.index(scope)
+        valtypes = dict.fromkeys(argnames, "funcargs")
         if not indirect:
             #XXX should we also check for the opposite case?
             for arg in argnames:
                 if arg not in self.fixturenames:
                     raise ValueError("%r uses no fixture %r" %(
                                      self.function, arg))
-        valtype = indirect and "params" or "funcargs"
+        else:
+            if not isinstance(indirect, (tuple, list)):
+                valtypes = dict.fromkeys(argnames, "params")
+            else:
+                for arg in indirect:
+                    if arg not in argnames:
+                        raise ValueError("indirect: fixture %r doesn't exist" %(
```

Comment:
Following the ValueError example above, I think it is useful to put the function name in the error message